src/Cake.FileHelpers/FileHelpers.cs

```diff
@@ -741,6 +729,44 @@ public static Group FindRegexMatchGroupInFile(this ICakeContext context, FilePat
 
             return null;
         }
+
+
+        private static StreamReader CreateStreamReader(ICakeContext context, FilePath file, Encoding? encoding = null)
+        {
+            var stream = context.FileSystem.GetFile(file).OpenRead();
+            return encoding is null
+                ? new StreamReader(stream, leaveOpen: false)
+                : new StreamReader(stream, encoding, leaveOpen: false);
+        }
+
+        private static StreamWriter CreateStreamWriter(ICakeContext context, FilePath file, FileMode mode, Encoding? encoding = null)
```

Comment:
```suggestion
        private static StreamWriter CreateStreamWriter(ICakeContext context, FilePath file, FileMode mode, Encoding encoding = null)
```

We don't yet support nullable references in this addin, so let's remove these for now.